src/Microsoft.Diagnostics.Monitoring.WebApi/Utilities/LogsUtilities.cs

```diff
@@ -53,6 +56,7 @@ public static async Task CaptureLogsAsync(TaskCompletionSource<object> startComp
             }
 
             await runTask;
+            await bufferingStream.DrainBufferAsync(outputStream, token);
```

Comment:
This changes the existing behavior that logs/metrics are written to the response stream as they are generated to buffering everything and flushing it all at once at the end. This could prevent scenarios such as infinite logging/metrics sessions to monitor for a specific log message of metrics pattern.

Is there some way we can have a stream or pipeline receive from the StreamingLoggerProvider, buffer to disk when it reaches some threshold, but allow simultaneous reading out to write to the response stream?

Reply:
Maybe something like: https://docs.microsoft.com/en-us/dotnet/standard/io/pipelines#pipe-basic-usage